src/kubernetes_charm.py

```diff
@@ -153,10 +172,74 @@ def _patch_service(self) -> None:
         )
         logger.debug("Patched k8s service")
 
+    def _get_node_name(self) -> str:
+        """Return the node name for this unit's pod ip."""
+        pod = lightkube.Client().get(
+            lightkube.resources.core_v1.Pod,
+            name=self.unit.name.replace("/", "-"),
+            namespace=self._namespace,
+        )
+        return pod.spec.nodeName
+
     # =======================
     #  Handlers
     # =======================
 
+    def get_all_k8s_node_hostnames_and_ips(
+        self,
+    ) -> typing.Tuple[typing.List[str], typing.List[str]]:
+        """Return all node hostnames and IPs registered in k8s."""
```

Comment:
@carlcsaposs-canonical this is to build the SAN part of the certificate. If you are the external client, you will see any of the worker nodes instead of the mysql-router-k8s unit. That means, any certificate must contain the IPs and hostnames of the existing workers.